corehq/apps/userreports/indicators/__init__.py

```diff
@@ -104,32 +104,34 @@ def get_values(self, item, context=None):
 
 class LedgerBalancesIndicator(ConfigurableIndicator):
 
-    def __init__(self, spec):
+    def __init__(self, spec, domain):
         self.product_codes = spec.product_codes
         self.column_id = spec.column_id
         self.ledger_section = spec.ledger_section
         self.case_id_expression = spec.get_case_id_expression()
+        self.domain = domain
         super(LedgerBalancesIndicator, self).__init__(spec.display_name)
 
     def _make_column(self, product_code):
         column_id = '{}_{}'.format(self.column_id, product_code)
         return Column(column_id, TYPE_INTEGER)
 
     @staticmethod
-    def _get_values_by_product(ledger_section, case_id, product_codes):
+    def _get_values_by_product(ledger_section, case_id, product_codes, domain):
         """returns a defaultdict mapping product codes to their values"""
-        values_by_product = StockState.objects.filter(
-            section_id=ledger_section,
-            case_id=case_id,
-            sql_product__code__in=product_codes,
-        ).values_list('sql_product__code', 'stock_on_hand')
-        return defaultdict(lambda: 0, values_by_product)
+        ret = defaultdict(lambda: 0)
+        ledgers = LedgerAccessors(domain).get_ledger_values_for_case(case_id)
+        for ledger in ledgers:
+            if ledger.section_id == ledger_section and ledger.entry_id in product_codes:
+                ret[ledger.entry_id] = ledger.stock_on_hand
+
+        return ret
 
     def get_columns(self):
         return map(self._make_column, self.product_codes)
 
     def get_values(self, item, context=None):
         case_id = self.case_id_expression(item)
-        values = self._get_values_by_product(self.ledger_section, case_id, self.product_codes)
+        values = self._get_values_by_product(self.ledger_section, case_id, self.product_codes, self.domain)
```

Comment:
or alternatively you could update the other references to follow this pattern. not sure whether there's any significant upside/downside of doing this in `FactoryContext` versus `EvaluationContext`. just thinking it would be nice if it were consistent across things.